Adds support for 'token' in email templates

pahrohfit opened the pull request:
Simple add of the token to be available in the email templates.

When using with an application via REST, being able to integrate with the device native app, using something like Branch.io.  This will allow a click via email to open either the native app, app store, or web site, to perform the function using the token from the request.

Commit:
Adds support for 'token' in email templates to assist with REST and Application based functions

jirikuncar reviewed (changes requested):
Can you please add a new test to `tests/test_context_processors.py` to make sure that the token is correctly passed to all mail templates? Thanks!

Commit:
Changed occurances of `encrypt_password` with `hash_password` due to depreciation warnings

Commit:
Fixed to do proper unit testing -- use a mock database connection.  Requires `mongomock` for MongoDB tests

Only a sociopath connects to an actual DB and does changes for unit testing a module.

Commit:
Included unit test for `{{ token }}` in welcome email

Commit:
Included text based template test file for`{{ token }}` in welcome email
